controllers/volumereplicationgroup_controller.go

```diff
@@ -40,6 +52,96 @@ type VolumeReplicationGroupReconciler struct {
 	Scheme *runtime.Scheme
 }
 
+func getPredicateFunc() predicate.Funcs {
+	// predicate functions send reconcile requests for create and delete events.
+	// For them the filtering of whether the pvc belongs to the any of the
+	// VolumeReplicationGroup CRs and identifying such a CR is done in the
+	// map function by comparing namespaces and labels.
+	// But for update of pvc, the reconcile request should be sent only for
+	// spec changes. Do that comparison here.
+	pvcPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
```

Comment:
(nit) barring the additional log message, we can remove this predicate, as the default predicate returns `true` and should suffice?

Reply:
OK. IIUC you are telling me to not have the "CreateFunc:" as it is only logging the event and then doing what would be the default behavior (return true).

This is my thinking. This is with my limited exposure to go. Let me know if I am missing something.

1) Having the log message helps in understanding the events happened better. Especially in testing the situations where a pvc is created for an application not disaster protected (i.e. a VolumeReplicationGroup CR does not exist for that application). The log helps in verifying that the create event is received, but reconcile is not sent. But, if I remove the predicate "CreateFunc", then we wont be able to log it at appropriate situation (which is the create event). We can use generic predicate, but, again we will only want that log to appear for the specific event of create.

2) I tried removing "return true" from the predicate as that is the default behavior. However with that lint started complaining saying a return statement is not there.

But if you think that logging is not needed (or not that helpful), then I can remove the entire "CreateFunc" predicate. As I said above, my only reason to have the CreateFunc predicate is to be able to log the event. 

Let me know your thoughts.

Reply:
So this is currently present only for the log afaict.

I was looking at removing the entire predicate function, and was expecting that the reconciler will be queued with the appropriate VRG for the PVC. Does that work? just removing the `return true` will not work as the function template expects a `bool` return.

If the above works, for now add a comment to state this is purely for debugging, or change the log to `log.Debug`

I suggest we add a common log message in `filterPVC` to state which VRG was queued, due to which PVC (or which PVC was ignored etc.). (maybe later?)

Reply:
> So this is currently present only for the log afaict.
> 

Yes. That is correct. It helped me while adding pvc watcher functionality to clearly distinguish situations like.

1) New pvc being created for the application which is disaster protected (i.e. has a VRG CR for it)
2) New pvc being created for the application which is not disaster protected (i.e. tested it by creating a new application in a
     different namespace for which VRG CR is not there).

For (1) there was only the log message saying a create event is received. For (2), there was log message and also the output of the reconcile logic.

> I was looking at removing the entire predicate function, and was expecting that the reconciler will be queued with the appropriate VRG for the PVC. Does that work? just removing the `return true` will not work as the function template expects a `bool` return.
> 
> If the above works, for now add a comment to state this is purely for debugging, or change the log to `log.Debug`
> 
Yes, the method suggested works. i.e. removing the "CreateFunc" predicate by default returns true which means the
reconcile request would be sent to the reconciler. 

ACK. I will add a comment saying the reason for "CreateFunc" predicate to exist is mainly for logging and debugging purposes and change the log level to Debug.

> I suggest we add a common log message in `filterPVC` to state which VRG was queued, due to which PVC (or which PVC was ignored etc.). (maybe later?)

There is a log message in filterPVC function where it logs when the matching VRG CR is found (i.e. VRG CR from the same namespace as that of the pvc whose label selector matches the label on pvc). But I agree with you that more information can be added in the log message such as name of the VRG CR, name of the PVC. Like you suggested, I can send this change as a separate PR.

Reply:
> ACK. I will add a comment saying the reason for "CreateFunc" predicate to exist is mainly for logging and debugging purposes and change the log level to Debug.

Done, let's get this addressed and merged.